optimum/intel/neural_compressor/quantization.py

```diff
@@ -271,15 +281,19 @@ def from_pretrained(
 
             q_model_name = q_model_name if q_model_name is not None else WEIGHTS_NAME
             revision = download_kwargs.pop("revision", None)
-            if os.path.isdir(model_name_or_path):
-                state_dict_path = os.path.join(model_name_or_path, q_model_name)
-            elif os.path.isfile(model_name_or_path):
-                state_dict_path = model_name_or_path
-            else:
-                state_dict_path = hf_bucket_url(model_name_or_path, filename=q_model_name, revision=revision)
+            if _use_cached_file == False:
+                if os.path.isdir(model_name_or_path):
+                    state_dict_path = os.path.join(model_name_or_path, q_model_name)
+                elif os.path.isfile(model_name_or_path):
+                    state_dict_path = model_name_or_path
+                else:
+                    state_dict_path = hf_bucket_url(model_name_or_path, filename=q_model_name, revision=revision)
```

Comment:
I suggest using `hf_hub_download` as follows :
```
  if os.path.isdir(model_name_or_path):
      state_dict_path = os.path.join(model_name_or_path, q_model_name)
  elif os.path.isfile(model_name_or_path):
      state_dict_path = model_name_or_path
  else:
      try:
          state_dict_path = hf_hub_download(model_name_or_path, filename=q_model_name, revision=revision)
      except EnvironmentError as err:
          logger.error(err)
          msg = (
              f"Can't load config for '{model_name_or_path}'. Make sure that:\n\n - '{model_name_or_path}' is a "
              f"correct model identifier listed on 'https://huggingface.co/models'\n\n - or "
              f"'{model_name_or_path}' is a correct path to a directory containing a {q_model_name} file\n\n"
          )

          if revision is not None:
              msg += (
                  f"- or '{revision}' is a valid git identifier (branch name, a tag name, or a commit id)  "
                  f"thatexists for this model name as listed on its model page on "
                  f"'https://huggingface.co/models'\n\n"
              )

          raise EnvironmentError(msg)
```

Reply:
hi， @echarlaix ， I see  cached_file called hf_hub_download as well. meanwhile, in cached_file, there's a judgement of offline mode . in some case, we need to use offline mode.   if we use hf_hub_download, should we move the judgement of offline mode here?

Reply:
I agree with you, that would be great

Reply:
so, my suggestion is to keep it as what this PR did, since both old cached_path and new cached_file API contain the logic of how to dispose offline mode.

Reply:
My suggestion would be to add the `local_files_only` variable instead. Currently in the PR, the behavior is not strictly the same depending on the version of transformers (`config_name_or_path` should not be the path of the file but only the directory containing it when using `cached_file`) also the addition of exception to handle the different imports might add additional difficulties in the future to maintain or modify the code. What do you think ?

Reply:
Hi, echarlaix, that means we need to port nearly all the logic in function "cached_file" here, including offline mode check, fetch local cache logic, maybe some function need to be added in __init__.py in transformer/utils like try_to_load_from_cache

Reply:
In our case can't we just add the offline mode detection (wtih the `local_files_only` variable) ?

Reply:
Hi, Echarlaix, there are three cases here
1) the config file is in local. like current distillation.cfg in example . need to add logic if it's local file here.
2) the config is in huggingface.co/models, not downloaded yet. need to use hf_hub_download to download to cache
3) there's a download cache in local. the config is there. no need to use hf_hub_download if it's offline mode, just try_to_load_from_cache. the function is very useful when we test in no network environment.

I thinks we need to support all the scenario here as transformers.

Reply:
Hi @sywangyi, for me all those 3 cases would be supported, can you go into more details about which one would not be handled correctly ? Also when loading a config from the cache `hf_hub_download` will be used (and not `try_to_load_from_cache`) as `_commit_hash` will be `None`

Reply:
hi, echarlaix, if __commit_hash will be None in the case, nothing concern from me. just update the patch according to your suggestion and add cache, offline support